src/qibolab/result.py

```diff
@@ -1,100 +1,91 @@
-from dataclasses import dataclass
 from functools import cached_property
 from typing import Optional
 
 import numpy as np
 import numpy.typing as npt
 
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64)])
+iq_record_dtype = np.dtype([("i", np.float64), ("q", np.float64)])
 
 
-@dataclass
```

Comment:
Dataclasses are definitely not mandatory, but they are a convenient approach:
- there is a simple declaration of the content, so it improves transparency (here it is not so immediate to understand which are the attributes available)
- it is automatically defining some extra methods
- it gives some further tools (like `fields` reflection, and automated serialization with `asdict`), that otherwise might be implemented over and over manually

However, there are contexts in which they are more inconvenient, and a normal class is more suitable. Especially when the class is not a data container.
But this case seems the exact opposite: `ExecutionResults` is supposed to be a data container, so I would recommend keeping it. But if you have a different opinion, I'd be interested in understanding the motivations.